sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_internal/async_polling.py

```diff
@@ -0,0 +1,84 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import base64
+from typing import TYPE_CHECKING
+
+from azure.core.polling.async_base_polling import AsyncLROBasePolling
+
+if TYPE_CHECKING:
+    from typing import Optional
+
+
+class KeyVaultAsyncBackupClientPollingMethod(AsyncLROBasePolling):
+    def initialize(self, client, initial_response, deserialization_callback):
+        """Set the initial status of this LRO.
+
+        :param initial_response: The initial pipeline response of the poller, or a URL continuation token
+        :raises: HttpResponseError if initial status is incorrect LRO state
+        """
+        if isinstance(initial_response, str):
```

Comment:
Yep, that's the case. I haven't found a way to pass a PipelineResponse `initial_response` to intialialize() when we're working from a continuation token -- the only I think we could, off the top of my head, would be to first make a GET request to the URL in from_continuation_token(). I initially had this initial status request in initialize(), but ultimately thought it made more sense to keep the string `initial_response` since a similar GET request would be made when we start polling

Reply:
Passing in an actual PipelineResponse would possibly make things a lot easier, since many or all other overrides wouldn't be necessary. The only issues I see with it would be that 1. we're making an extra request (not too big a deal), and 2. our "initial response" at that point would actually be a status update from the moment of rehydration, rather than the expected status from the poll before the continuation token was obtained (possibly a bigger deal)

Reply:
For record-keeping, I should note here that the biggest problem with making a status request in from_continuation_token() is that it would require making an async pipeline request, inside a synchronous classmethod, for the async polling method